lib/auth/trustedcluster.go

```diff
@@ -67,29 +119,27 @@ func (a *Server) UpsertTrustedCluster(ctx context.Context, tc types.TrustedClust
 
 	// if there is no existing cluster, switch to the create case
 	if existingCluster == nil {
-		return a.createTrustedCluster(ctx, tc)
+		return a.createTrustedCluster(ctx, tc, validateName)
 	}
 
-	if err := existingCluster.CanChangeStateTo(tc); err != nil {
-		return nil, trace.Wrap(err)
-	}
+	updated, err := a.updateTrustedCluster(ctx, tc, existingCluster)
+	return updated, trace.Wrap(err)
```

Comment:
Should we give this a handful of tries? UpsertFoo should basically never fail due to concurrent writes, but this logic is liable to fail on a race between an upsert and a delete, or two upserts on a new trusted cluster.

Reply:
That is a good point. I think we can defer this to a separate PR though. We'd also be modifying the behavior of the existing Upsert method.